File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/permission/PermissionsGeneralDetailsFragment.java
package sk.styk.martin.apkanalyzer.activity.permission;


import android.content.Context;
import android.content.pm.PackageManager;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import sk.styk.martin.apkanalyzer.databinding.FragmentPermissionDetailGeneralBinding;
import sk.styk.martin.apkanalyzer.model.detail.PermissionData;

/**
 * Created by Martin Styk on 13.01.2018.
 */
public class PermissionsGeneralDetailsFragment extends Fragment {

    private static final String TAG = PermissionsGeneralDetailsFragment.class.getSimpleName();

    public static final String ARG_NUMBER_GRANTED_APPS = "apps_granted_perm";
    public static final String ARG_NUMBER_NOT_GRANTED_APPS = "apps_not_granted_perm";

    private FragmentPermissionDetailGeneralBinding binding;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentPermissionDetailGeneralBinding.inflate(getLayoutInflater());

        PermissionData permissionData = getArguments().getParcelable(PermissionDetailPagerFragment.ARG_CHILD);
        binding.setPermission(permissionData);
        binding.setContext(getContext());
        binding.setGrantedApps(getArguments().getInt(ARG_NUMBER_GRANTED_APPS));
        binding.setNotGrantedApps(getArguments().getInt(ARG_NUMBER_NOT_GRANTED_APPS));

        return binding.getRoot();
    }

}

File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/permission/PermissionsGeneralDetailsFragment.kt
package sk.styk.martin.apkanalyzer.activity.permission


import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import sk.styk.martin.apkanalyzer.R
import sk.styk.martin.apkanalyzer.databinding.FragmentPermissionDetailGeneralBinding

/**
 * @author Martin Styk
 * @version 15.01.2017
 */
class PermissionsGeneralDetailsFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        retainInstance = true
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding: FragmentPermissionDetailGeneralBinding = DataBindingUtil
                .inflate(inflater, R.layout.fragment_permission_detail_general, container, false)

        binding.data = arguments.getParcelable(PermissionDetailPagerFragment.ARG_CHILD) ?: throw IllegalArgumentException("data null")
        binding.granted = arguments.getInt(ARG_NUMBER_GRANTED_APPS)
        binding.notGranted = arguments.getInt(ARG_NUMBER_NOT_GRANTED_APPS)

        return binding.root
    }

    companion object {
        const val ARG_NUMBER_GRANTED_APPS = "apps_granted_perm"
        const val ARG_NUMBER_NOT_GRANTED_APPS = "apps_not_granted_perm"
    }

}
